src/Dispatcher.php

```diff
@@ -52,6 +54,16 @@ public function addResponseMiddleware(callable $callback): self
         return $this;
     }
 
+    public function setAuthenticationProvider(Interfaces\AuthenticationProviderInterface $provider)
```

Comment:
It might be preferable to combine these two setters into one, requiring either both or neither. It will rarely make sense to authenticate without authorizing, and a misconfiguration could result in resource protection failing open.